table/block_based/block_based_table_iterator.cc

```diff
@@ -303,12 +308,22 @@ bool BlockBasedTableIterator::NextAndGetResult(IterateResult* result) {
 }
 
 void BlockBasedTableIterator::Prev() {
-  // Return Error.
-  if (readahead_cache_lookup_) {
-    block_iter_.Invalidate(
-        Status::NotSupported("auto tuning of readahead_size in is not "
-                             "supported with Prev operation."));
-    return;
+  if (readahead_cache_lookup_ && !IsIndexAtCurr()) {
+    // In case of readahead_cache_lookup, db_iter calls SeekForPrev for Prev
+    // operation, to reverse the direction and disable readahead_cache_lookup.
+    // However, there can be cases SeekForPrev couldn't be called. So, It'll
+    // call SeekForPrev here in that case to point index_iter_ to the right
+    // block.
+    if (!last_key_.empty()) {
+      SeekForPrev(last_key_);
+    }
+
+    if (last_key_.empty() || !Valid()) {
+      block_iter_.Invalidate(
```

Comment:
Now that the auto_readahead_size option is default on, will this result in an error status returned to the user?

Reply:
It's not enabled but when enabled, this error is returned to the user.

Reply:
Why is it necessary to invalidate with error status? For backward iteration, can we just ignore/clear the auto_readahead_size option? Returning an error is unnecessarily heavy handed I think.